code/game/machinery/computer/pod.dm

```diff
@@ -1,21 +1,38 @@
 /obj/machinery/computer/pod
 	name = "mass driver launch control"
 	desc = "A combined blastdoor and mass driver control unit."
+	/// Connected mass driver
 	var/obj/machinery/mass_driver/connected = null
-	var/title = "Mass Driver Controls"
+	/// ID of the launch control
 	var/id = 1
-	var/timing = 0
+	/// If the launch timer counts down
+	var/timing = FALSE
+	/// Time before auto launch
 	var/time = 30
+	/// Range in which we search for a mass drivers and poddoors nearby
 	var/range = 4
 
-
 /obj/machinery/computer/pod/Initialize()
 	. = ..()
 	for(var/obj/machinery/mass_driver/M in range(range, src))
 		if(M.id == id)
 			connected = M
+			break
 
+/obj/machinery/computer/pod/process()
+	if(!..())
+		return
+	if(timing)
+		if(time > 0)
+			time = round(time) - 1
+		else
+			alarm()
+			time = 0
+			timing = FALSE
```

Comment:
Alarm `sleep()`s, so process() becomes async and you get multiple concurrent process() procs running in parallel before `timing` is set to FALSE by one of them, leading to the Mass Driver constantly triggering when it should only activate once.